modules/generator/registry/registry.go

```diff
@@ -208,20 +240,48 @@ func (r *ManagedRegistry) onAddMetricSeries(count uint32) bool {
 	return true
 }
 
-func (r *ManagedRegistry) onRemoveMetricSeries(count uint32) {
-	r.activeSeries.Sub(count)
+func (r *ManagedRegistry) onUpdateEntity(hash uint64) {
+	r.updatedSinceLastCollectionMtx.Lock()
+	r.updatedSinceLastCollection[hash] = struct{}{}
+	r.updatedSinceLastCollectionMtx.Unlock()
+}
 
+func (r *ManagedRegistry) onRemoveEntity(count uint32) {
+	r.activeSeries.Sub(count)
 	r.metricTotalSeriesRemoved.Add(float64(count))
 }
 
 func (r *ManagedRegistry) CollectMetrics(ctx context.Context) {
+	r.updatedSinceLastCollectionMtx.RLock()
+	hashes := maps.Clone(r.updatedSinceLastCollection)
+	r.updatedSinceLastCollection = make(map[uint64]struct{})
+	r.updatedSinceLastCollectionMtx.RUnlock()
+
 	r.metricsMtx.RLock()
 	defer r.metricsMtx.RUnlock()
 
+	rejected, err := r.entityLimiter.TrackEntities(ctx, r.tenant, slices.Collect(maps.Keys(hashes)))
+	if err != nil {
+		r.limitLogger.Log("msg", "tracking entities failed", "err", err)
+	}
+
+	if len(rejected) > 0 {
+		r.limitLogger.Log("msg", "max active entities reached", "rejected", len(rejected), "updated_since_last_collection", len(hashes))
+	}
+
+	rejectedMap := make(map[uint64]struct{})
```

Comment:
i'm still digging on what this deleteFunc does but why not have TrackEntities return a `map[uint64]struct{}` to save allocs?